src/inmanta/server/bootloader.py

```diff
@@ -164,38 +168,49 @@ def _load_extension(self, name: str) -> Callable[[ApplicationContext], None]:
             raise PluginLoadFailed(f"Could not load module {name}") from e
 
         try:
-            mod = importlib.import_module(f"{name}.{EXTENSION_MODULE}")
-            return mod.setup
+            return importlib.import_module(f"{name}.{EXTENSION_MODULE}")
         except Exception as e:
             raise PluginLoadFailed(f"Could not load module {name}.{EXTENSION_MODULE}") from e
 
-    def _load_extensions(self) -> Dict[str, Callable[[ApplicationContext], None]]:
+    def _load_extensions(self, load_all_extensions: bool = False) -> Dict[str, ModuleType]:
         """Discover all extensions, validate correct naming and load its setup function"""
-        plugins: Dict[str, Callable[[ApplicationContext], None]] = {}
-        for name in self._discover_plugin_packages():
+        plugins: Dict[str, ModuleType] = {}
+        for name in self._discover_plugin_packages(load_all_extensions):
             try:
-                plugin = self._load_extension(name)
+                module = self._load_extension(name)
                 assert name.startswith(f"{EXTENSION_NAMESPACE}.")
                 name = name[len(EXTENSION_NAMESPACE) + 1 :]
-                plugins[name] = plugin
+                plugins[name] = module
             except PluginLoadFailed:
                 LOGGER.warning("Could not load extension %s", name, exc_info=True)
         return plugins
 
-    # Extension loading Phase II: collect slices
-    def _collect_slices(self, extensions: Dict[str, Callable[[ApplicationContext], None]]) -> ApplicationContext:
+    def _collect_slices(self, ext_name: str, ext_module: ModuleType, app_ctx: ApplicationContext) -> None:
+        try:
+            ext_module.setup(app_ctx)
+        except Exception:
+            LOGGER.warning("Could not load extension %s", ext_name, exc_info=True)
+
+    def _collect_environment_settings(self, ext_name: str, ext_module: ModuleType, app_ctx: ApplicationContext) -> None:
+        try:
+            ext_module.register_environment_settings(app_ctx)
+        except AttributeError:
+            # Extension doesn't defined any environment settings.
+            pass
+        except Exception:
+            LOGGER.warning("Could not load extension %s", ext_name, exc_info=True)
+
+    def load_slices(
+        self, load_all_extensions: bool = False, only_register_environment_settings: bool = False
```

Comment:
Since there is no logical ordering here, enforce readability by making them kwarg-only?
```suggestion
        self, *, load_all_extensions: bool = False, only_register_environment_settings: bool = False
```